spyder_kernels/console/kernel.py

```diff
@@ -311,11 +311,7 @@ def publish_pdb_state(self, step):
         Publish Variable Explorer state and Pdb step through
         send_spyder_msg.
         """
-        state = dict(
-            namespace_view=self.get_namespace_view(),
-            var_properties=self.get_var_properties(),
-            step=step
-         )
+        state = dict(step=step)
```

Comment:
Now that `publish_pdb_state` doesn't use any kernel function I don't think it should stay in kernel. I would move the logic here in `notify_spyder`. 

You can replace `self.frontend_call(blocking=False)` by `frontend_request(blocking=False)`